### PR TITLE
Architecture tests initialization 

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | -----
| Branch?         | 1.10
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

This is my first try to use [PHPArkitect](https://github.com/phparkitect/arkitect) tool - but it seems promising! I believe with this library we could ensure some architectural conventions and proper dependencies. Let me know what you think about it 🖖 🚀 🏗️

### CODE REVIEW

File: phparkitect.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+use Arkitect\Rules\Rule;
+use PhpSpec\ObjectBehavior;
+
+return static function (Config $config): void
+{
+    $srcClassSet = ClassSet::fromDir(__DIR__.'/src');
+
+    $rules = [];
+
+    $rules[] = Rule::allClasses()
```
Comment (pjedrzejewski) [inline phparkitect.php:20]: Uuu, me likes! 🧠

File: .github/workflows/application.yml [hunk 2]
```diff
@@ -109,6 +109,42 @@ jobs:
...
+        runs-on: ubuntu-latest
+
+        name: "Architecture tests (PHPArkitect)"
+
+        timeout-minutes: 10
+
+        strategy:
+            fail-fast: false
+            matrix:
+                php: ["7.4", "8.0"]
```
Comment (lchrusciel) [inline .github/workflows/application.yml:122]: Do we really need to test it for several PHP versions? What would you say for testing it only with the top one?

**Review by lchrusciel — APPROVED** [review]

File: .github/workflows/application.yml (hunk 2 above)
Comment (Zales0123) [reply .github/workflows/application.yml:122]: In fact... we don't 🖖

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Mateusz! :1st_place_medal:

**Comment by pfazzi** [comment]
Happy to see you started using PHPArkitect! 🥳

**Pull request merged**